public/app/features/plugins/extensions/registry/ExportedComponentsRegistry.test.ts

```diff
@@ -1,15 +1,62 @@
 import React from 'react';
 import { firstValueFrom } from 'rxjs';
 
+import { PluginLoadingStrategy } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+import { isGrafanaDevMode } from '../utils';
+
 import { ExposedComponentsRegistry } from './ExposedComponentsRegistry';
 import { MSG_CANNOT_REGISTER_READ_ONLY } from './Registry';
 
+jest.mock('../utils', () => ({
+  ...jest.requireActual('../utils'),
+
+  // Manually set the dev mode to false
+  // (to make sure that by default we are testing a production scneario)
+  isGrafanaDevMode: jest.fn().mockReturnValue(false),
+}));
+
 describe('ExposedComponentsRegistry', () => {
   const consoleWarn = jest.fn();
+  const originalApps = config.apps;
+  const pluginId = 'grafana-basic-app';
+  const appPluginConfig = {
+    id: pluginId,
+    path: '',
+    version: '',
+    preload: false,
+    angular: {
+      detected: false,
+      hideDeprecation: false,
+    },
+    loadingStrategy: PluginLoadingStrategy.fetch,
+    dependencies: {
+      grafanaVersion: '8.0.0',
+      plugins: [],
+      extensions: {
+        exposedComponents: [],
+      },
+    },
+    extensions: {
+      addedLinks: [],
+      addedComponents: [],
+      exposedComponents: [],
+      extensionPoints: [],
+    },
+  };
```

Comment:
Since this mock is being used in a few places, maybe worth adding it to a `__mocks__` dir?

Reply:
I think that's a good point 👍

Reply:
Hmm, actually, I am being a little bit puzzled here. How are we using the `__mocks__` dir usually? If we use it for manual Jest mocks (mocking a module) then I guess this wouldn't really fit there? (It's not a module, but part of the config data structure). Or would you imagine just putting it in a JSON file and requiring it in from the different test cases that depend on it?

Reply:
Ah that's true. Moving it to its own file sounds like a good idea. 👍